src/server/routes/editor.tsx

```diff
@@ -179,18 +180,31 @@ router.post('/edit/handler', auth.isAuthenticatedForHandler, (req, res) => {
 });
 
 router.post('/privs/edit/handler', auth.isAuthenticatedForHandler, auth.isAuthorized(ADMIN), async (req, res, next) => {
-	const {Editor} = req.app.locals.orm;
+	const {Editor, AdminLog, bookshelf} = req.app.locals.orm;
+	const {adminId, newPrivs, note, oldPrivs, targetUserId} = req.body;
+	const actionType = AdminActionType.CHANGE_PRIV;
 	try {
+		const trx = await bookshelf.transaction();
 		const editor = await Editor
-			.forge({id: parseInt(req.body.targetUserId, 10)})
-			.fetch({require: true})
+			.forge({id: parseInt(targetUserId, 10)})
+			.fetch({require: true, transacting: trx})
 			.catch(Editor.NotFoundError, () => next(new error.NotFoundError('Editor not found', req)));
-		if (editor.get('privs') === req.body.newPrivs) {
+		if (editor.get('privs') === newPrivs) {
 			return next(new error.FormSubmissionError(
 				'No change to Privileges', req
 			));
 		}
-		await editor.save({privs: req.body.newPrivs});
+		await editor.save({privs: newPrivs});
+		const logData: AdminLogDataT = {
+			actionType,
+			adminId,
+			newPrivs,
+			note,
+			oldPrivs,
+			targetUserId
+		};
+		await createAdminLog(logData, AdminLog, trx);
+		await trx.commit();
```

Comment:
This is unnecessary actually, my bad. From the [docs](https://bookshelfjs.org/api.html#Bookshelf-instance-transaction):

> Note that there is no need to explicitly call transaction.commit() or transaction.rollback() since the entire transaction will be committed if there are no errors inside the transaction block.

Reply:
I think we'll have to do this. This documentation is perhaps only for when we are using transaction in the form of a block as shown in the example. I am getting errors if I remove this.